### DISCUSSION TITLE
Shiny texture

### DISCUSSION THREAD

**Discussion by Rhagven** [TokisanGames/Terrain3D#327] [Questions] [answered]
Hey, I'm very much a novice to game design, and your godot addon has been insanely helpful, but some of the textures I use to paint the terrain ends up extremely shiny.  I followed the instructions to pack them in GIMP, and I used the "paint wetness" brush at 100% to try and make them less reflective, and I even increased the brightness of the roughness file and repacked it into a new normal texture (to try and make the white areas more white to increase roughness, though I don't know if that's how it works) but nothing seemed to help.  I was messing with the settings in the DirectionalLight3D node to see if there was anything there that would help, but it didn't seem to, so my question is "Is there anything aside from paint wetness that would allow me to increase the roughness or decrease reflectivity of my textures in your addon?  Or is there something I'm missing that may be making some of my textures very shiny while others are matte?"

Here's an example of what I'm talking about, there are two textures painted on the terrain, but when the dirt texture is in between the player and the light source it becomes very reflective.  Apologies if this isn't the right place to ask this.

![shiny](https://github.com/TokisanGames/Terrain3D/assets/161252244/2f7996a8-f047-4668-85d7-22a1d53588ec)

**Comment by TokisanGames** [comment]
> Or is there something I'm missing that may be making some of my textures very shiny while others are matte?

If the demo textures look normal, then yes.

Show a picture of your roughness texture (alpha layer), and normal texture(rgb). And what the combined normal/roughness file looks like in the inspector in Godot when you double click the file. 


Regarding your picture, it's too dark to see anything.

**Reply by Rhagven** [reply]
Sorry, I tried a few times to get a better screen capture, but it kept showing up way darker than it looked on my end so I thought it was just my settings.  Here is a side by side of the normal (GL) texture and the roughness, the inspector window when I double click the packed version, and a (hopefully) more helpful screen capture of the in-game look.  The two circled portions are the same texture but the one on the right just appears so much more reflective.
![normal + roughness](https://github.com/TokisanGames/Terrain3D/assets/161252244/4d84e19a-45c7-415b-8b76-055974ad1b30)
![inspector](https://github.com/TokisanGames/Terrain3D/assets/161252244/a0d3adea-5f34-47d4-93ab-e85f624a8abd)
![shiny](https://github.com/TokisanGames/Terrain3D/assets/161252244/12b0b6e4-2cee-4ceb-b944-11e4aedd41ba)

**Reply by TokisanGames** [reply]
What versions of Godot and Terrain3D?

OS and card?

What happened to the environment? What does it look like under the default demo environment and directional light?

Are you using a custom shader? 

Is wetness / roughness modification erased everywhere? Use the debug view to confirm. 

The hill in the top left doesn't look right. Enable shadows on your light and show a more zoomed out picture of the terrain with hills. Looks like your normals are broken.

**Reply by Rhagven** [reply]
I'm on Godot v4.2.1.stable.official and terrain3D 0.9.1

OS: Windows 10 Home with a Radeon RX 6800 XT

The environment probably looks bad because it is one that I am trying to build, not the demo that came with the addon, haha!  I don't see the shiny issue in any of the demo textures, which is why I didn't think it was an issue with the addon, but something I was doing incorrectly.

I'm not using any custom shaders.  To be honest, I haven't learned enough to even try messing with that yet.

Wetness/Roughness was painted to 100% in a large area around where the character spawns in (which is where I took the screen capture from)
Here is the debug view, I tried to match the same view so you can see the geometry of the edge of the land in the top right to see that is the same area I was looking at.  I don't know if it matters, but I had the brush size, opacity, and roughness all set to the maximum and just painted the entire area with many sweeps just to ensure it covered everything, and this is the result.

![debug](https://github.com/TokisanGames/Terrain3D/assets/161252244/3d4e794c-c550-4f61-88ea-3938ae1280d2)

Just to reiterate I am very painfully new to this.  The shadows look like a hot dumpster fire, and I don't know what is going on there.  I haven't researched any of that yet to be honest, I was just trying to figure out why the dirt was so shiny, but here is a screen cap with shadows "on" in the directionalLight node

![shadows](https://github.com/TokisanGames/Terrain3D/assets/161252244/6b597ab0-0f92-4346-8490-4ab6e85056b0)

If my normals are broken, what does that mean, that I didn't pack them correctly?  Do I need to change some of the export settings to mess with them, or try using the DX normal instead of GL?  The only other thing I could find online was people saying to change the "specularity" but I didn't see that option anywhere.

**Reply by TokisanGames** [reply]
Which renderer are you using? It's in the top right corner. 

You have far too many changes. You cannot test anything like this. For this ticket only test in the demo. Show me a picture of the default demo terrain, lighting, and environment.

Then put your dirt texture in the demo as a third texture and paint a section. Don't use wetness. Angle it where light should reflect off of the ground and send a picture.

**Reply by Rhagven** [reply]
That is a great idea I should have thought of.  I'm using forward+ for the renderer, and here is a screenshot of the regular demo

![demo](https://github.com/TokisanGames/Terrain3D/assets/161252244/f4516576-f002-4149-81f3-9dd557ef33d8)

I tried to add the dirt texture as a third, but it completely messes everything up and gives an error message

![error](https://github.com/TokisanGames/Terrain3D/assets/161252244/839aa4ed-d8fc-4544-88fd-fd4fd18b3c98)

The main difference I notice off the bat is when I edit the original two, they have some kind of different label, maybe the compression?  I'm not sure, but here they are.

![difference](https://github.com/TokisanGames/Terrain3D/assets/161252244/ffdfad10-6c7c-4eb8-8c27-7e47d9f34b57)

Your original texture is on the left with the BPTC_RGBA label and the channel pack I exported following the directions has the DXT5 RGBA8 label.  

I appreciate your time, but I can see this is some weird issue that has nothing to do with your plugin and probably as you said is something wrong with the textures I made judging from this error.  I'm willing to keep trying to figure it out if you're willing to keep helping, but at the same time I'm sure you have better stuff to do than deal with this mess, hahaha.  For what it's worth, I thought the terrain editing part worked amazingly, and was very excited to use it.  I just hope I can create some workable textures to go with it!  haha

**Reply by TokisanGames** [reply]
> Your original texture is on the left with the BPTC_RGBA label and the channel pack I exported following the directions has the DXT5 RGBA8 label.

Mark high quality on the import panel for bptc.

> I appreciate your time, but I can see this is some weird issue that has nothing to do with your plugin and probably as you said is something wrong with the textures I made judging from this error

The problems you're experiencing are from lack of understanding and a messed up environment in your project. I haven't seen any thing wrong with your textures yet, except not matching the type for the demos.

**Reply by TokisanGames** [reply]
Did this fix it?

**Reply by Rhagven** [reply]
Sorry, some other things have kept me busy.  I haven't been able to try all of the tests I wanted, but to update so far.  
-The original channel packs I made were .DDS files, which I could not change any import options with those in Godot (which you probably already knew)
-I repacked them as .PNG following the instructions and added them to the demo in high quality.  They were still equally shiny initially, but then I was able to paint them with the wetness tool and in the DEMO setting at 100% roughness they looked a LOT better (still a tiny bit shiny under direct light, but very much improved over the texture in my custom map)

I haven't had the time to repack all of my textures as .PNG files (when I added just the one, it caused the same error as the demo which I assumed would happen, but I wanted to see just in case.)  But  I feel like that may end up being the best option for me to get them to be a bit more matte.  I will try to do that over the weekend just to update again in case anyone else is having some similar issue, maybe .PNG might be a better fit?  But I'll post screenshots to compare

**Reply by TokisanGames** [reply]
Where did you get this texture?

**Reply by Rhagven** [reply]
https://ambientcg.com/view?id=Ground048

I have used this site for all of my textures, but this is the one specifically for this issue.  I used the 1k_PNG format

**Reply by TokisanGames** [reply]
I setup that texture in the demo and only see a vague shine on it, similar to the rock above.

![image](https://github.com/TokisanGames/Terrain3D/assets/632766/7d51c980-961b-4b19-b357-d70779d957c1)

It's actually less shiny than on the source website, likely due to the specular reduction in the shader.

![image](https://github.com/TokisanGames/Terrain3D/assets/632766/c6482d4d-9040-4c85-bd46-98c70eb14cd0)

You most likely don't need to increase roughness, in fact I'll likely remove that feature in favor of something else. Instead look at your texture setup and especially your environment and lighting setup.

**Reply by TokisanGames** [reply, marked as answer]
Closing this due to discussion stopped, and there being no problem with Terrain3D or the texture provided. Both work as expected, without needing to use the roughness modifier. We can reopen if there are further issues.
